packages/vite/types/customEvent.d.ts

```diff
@@ -12,6 +12,8 @@ export interface CustomEventMap {
   'vite:beforeFullReload': FullReloadPayload
   'vite:error': ErrorPayload
   'vite:invalidate': InvalidatePayload
+  'vite:ws:connect': WebSocket
+  'vite:ws:disconnect': WebSocket
```

Comment:
Do you need the WebSocket instance? It concerns me a bit that WebSocket instance has a broad interface. (e.g. `socket.send()`)

Reply:
It isn't required for the original request. @antfu maybe we should add this param once we have a real use case? And once we go there, it could be a Payload object like the others so we could extend it in the future if needed.

Reply:
Good points, I think we can talk about this in the next team meeting 👍🏼 
I assume you also don't think it is a good idea to have a Payload object here.